tests/test_function_id.py

```diff
@@ -0,0 +1,24 @@
+import pytest
+
+import capa.features.insn
+
+from fixtures import pma16_01_extractor, get_function, extract_function_features
+
+
+def test_function_id_simple_match(pma16_01_extractor):
+    assert pma16_01_extractor.is_library_function(0x407490) == True
+    assert pma16_01_extractor.get_function_name(0x407490) == "__aulldiv"
+
+
+@pytest.mark.xfail
```

Comment:
i dont have a good signature yet that relies on a reference name to match. this will be easier once we have VS6 support. https://github.com/williballenthin/siglib/issues/2

Reply:
we do now 😉

Reply:
vivisect doesn't recognize any of the function VAs mentioned below as functions?

Reply:
```
python scripts/match-function-id.py tests/data/Practical\ Malware\ Analysis\ Lab\ 16-03.exe_ -d --signature sigs/flare_msvc_rtf_32_64.sig
DEBUG:capa:perf: flirt: parsing .sig: sigs/flare_msvc_rtf_32_64.sig: 0.71s
DEBUG:capa:perf: flirt: compiling sigs: 1.60s
DEBUG:capa.match-function-id:registering viv function analyzer: FlirtFunctionAnalyzer (sigs/flare_msvc_rtf_32_64.sig)
DEBUG:capa.match-function-id:matching function: 0x401a80
DEBUG:viv_utils.flirt:found local function name: 0x401a80: _xcptlookup
DEBUG:capa.match-function-id:matching function: 0x40193f
DEBUG:viv_utils.flirt:found library function: 0x40193f: __XcptFilter
DEBUG:capa.match-function-id:matching function: 0x401787
DEBUG:viv_utils.flirt:found local function name: 0x401787: __initterm
DEBUG:capa.match-function-id:matching function: 0x4016ee
DEBUG:viv_utils.flirt:found local function name: 0x4016ee: _doexit
DEBUG:capa.match-function-id:matching function: 0x4016cc
DEBUG:viv_utils.flirt:found library function: 0x4016cc: _exit
DEBUG:capa.match-function-id:matching function: 0x401060
DEBUG:capa.match-function-id:matching function: 0x4010e0
DEBUG:capa.match-function-id:matching function: 0x401300
DEBUG:capa.match-function-id:matching function: 0x401000
DEBUG:capa.match-function-id:matching function: 0x4017b0
DEBUG:viv_utils.flirt:found library function: 0x4017b0: _strncmp
DEBUG:capa.match-function-id:matching function: 0x4017f0
DEBUG:viv_utils.flirt:found library function: 0x4017f0: _strrchr
DEBUG:capa.match-function-id:matching function: 0x4011e0
DEBUG:capa.match-function-id:matching function: 0x4013d0
DEBUG:capa.match-function-id:matching function: 0x40169f
DEBUG:viv_utils.flirt:found library function: 0x40169f: __cinit
DEBUG:capa.match-function-id:matching function: 0x403cc4
DEBUG:viv_utils.flirt:found library function: 0x403cc4: ___old_sbh_release_region
DEBUG:capa.match-function-id:matching function: 0x403d1a
DEBUG:viv_utils.flirt:found library function: 0x403d1a: ___old_sbh_decommit_pages
DEBUG:capa.match-function-id:matching function: 0x403e33
DEBUG:viv_utils.flirt:found library function: 0x403e33: ___old_sbh_free_block
DEBUG:capa.match-function-id:matching function: 0x403ddc
DEBUG:viv_utils.flirt:found library function: 0x403ddc: ___old_sbh_find_block
DEBUG:capa.match-function-id:matching function: 0x404830
DEBUG:viv_utils.flirt:found local function name: 0x4048a0: ?
DEBUG:viv_utils.flirt:found local function name: 0x4048cc: ?
DEBUG:viv_utils.flirt:found local function name: 0x4048f0: ?
DEBUG:viv_utils.flirt:found local function name: 0x40492c: ?
DEBUG:viv_utils.flirt:found local function name: 0x404a28: ?
DEBUG:viv_utils.flirt:found local function name: 0x404a48: ?
DEBUG:viv_utils.flirt:found local function name: 0x404a70: ?
DEBUG:viv_utils.flirt:found local function name: 0x404ac4: ?
DEBUG:viv_utils.flirt:found library function: 0x404830: ?
DEBUG:capa.match-function-id:matching function: 0x4033a2
DEBUG:viv_utils.flirt:found library function: 0x4033a2: ___sbh_free_block
DEBUG:capa.match-function-id:matching function: 0x403377
DEBUG:viv_utils.flirt:found library function: 0x403377: ___sbh_find_block
DEBUG:capa.match-function-id:matching function: 0x4025d4
DEBUG:viv_utils.flirt:found library function: 0x4025d4: _free
DEBUG:capa.match-function-id:matching function: 0x402640
DEBUG:viv_utils.flirt:found library function: 0x402640: ?
DEBUG:viv_utils.flirt:found library function: 0x402650: ?
DEBUG:capa.match-function-id:matching function: 0x4016dd
DEBUG:viv_utils.flirt:found library function: 0x4016dd: __exit
DEBUG:capa.match-function-id:matching function: 0x4041a4
DEBUG:viv_utils.flirt:found library function: 0x4041a4: ___crtMessageBoxA
DEBUG:capa.match-function-id:matching function: 0x402650
DEBUG:capa.match-function-id:matching function: 0x404230
DEBUG:viv_utils.flirt:found library function: 0x404230: _strncpy
DEBUG:capa.match-function-id:matching function: 0x4027f0
DEBUG:viv_utils.flirt:found library function: 0x4027f0: _strlen
DEBUG:capa.match-function-id:matching function: 0x402481
DEBUG:viv_utils.flirt:found library function: 0x402481: __NMSG_WRITE
DEBUG:capa.match-function-id:matching function: 0x402448
DEBUG:viv_utils.flirt:found library function: 0x402448: __FF_MSGBANNER
DEBUG:capa.match-function-id:matching function: 0x4018f6
DEBUG:viv_utils.flirt:found library function: 0x4018f6: __amsg_exit
DEBUG:capa.match-function-id:matching function: 0x40432e
DEBUG:viv_utils.flirt:found library function: 0x40432e: __callnewh
DEBUG:capa.match-function-id:matching function: 0x404b70
DEBUG:viv_utils.flirt:found library function: 0x404b70: _memset
DEBUG:capa.match-function-id:matching function: 0x403b80
DEBUG:viv_utils.flirt:found library function: 0x403b80: ___old_sbh_new_region
DEBUG:capa.match-function-id:matching function: 0x404080
DEBUG:viv_utils.flirt:found library function: 0x404080: ___old_sbh_alloc_block_from_page
DEBUG:capa.match-function-id:matching function: 0x403e78
DEBUG:viv_utils.flirt:found library function: 0x403e78: ___old_sbh_alloc_block
DEBUG:capa.match-function-id:matching function: 0x403a85
DEBUG:viv_utils.flirt:found library function: 0x403a85: ___sbh_alloc_new_group
DEBUG:capa.match-function-id:matching function: 0x4039d4
DEBUG:viv_utils.flirt:found library function: 0x4039d4: ___sbh_alloc_new_region
DEBUG:capa.match-function-id:matching function: 0x4036cb
DEBUG:viv_utils.flirt:found library function: 0x4036cb: ___sbh_alloc_block
DEBUG:capa.match-function-id:matching function: 0x40276e
DEBUG:viv_utils.flirt:found library function: 0x40276e: __heap_alloc
DEBUG:capa.match-function-id:matching function: 0x402742
DEBUG:viv_utils.flirt:found library function: 0x402742: __nh_malloc
DEBUG:capa.match-function-id:matching function: 0x402730
DEBUG:viv_utils.flirt:found library function: 0x402730: _malloc
DEBUG:capa.match-function-id:matching function: 0x402a81
DEBUG:viv_utils.flirt:found local function name: 0x402a81: _setSBCS
DEBUG:capa.match-function-id:matching function: 0x403300
DEBUG:viv_utils.flirt:found library function: 0x403300: __alloca_probe
DEBUG:capa.match-function-id:matching function: 0x40456d
DEBUG:viv_utils.flirt:found local function name: 0x40456d: _strncnt
DEBUG:capa.match-function-id:matching function: 0x404349
DEBUG:viv_utils.flirt:found library function: 0x404349: ___crtLCMapStringA
DEBUG:capa.match-function-id:matching function: 0x404598
DEBUG:viv_utils.flirt:found library function: 0x404598: ___crtGetStringTypeA
DEBUG:capa.match-function-id:matching function: 0x402aaa
DEBUG:viv_utils.flirt:found local function name: 0x402aaa: _setSBUpLow
DEBUG:capa.match-function-id:matching function: 0x402a4e
DEBUG:viv_utils.flirt:found local function name: 0x402a4e: _CPtoLCID
DEBUG:capa.match-function-id:matching function: 0x402a04
DEBUG:viv_utils.flirt:found local function name: 0x402a04: _getSystemCP
DEBUG:capa.match-function-id:matching function: 0x40286b
DEBUG:viv_utils.flirt:found library function: 0x40286b: __setmbcp
DEBUG:capa.match-function-id:matching function: 0x402c2f
DEBUG:viv_utils.flirt:found library function: 0x402c2f: ___initmbctable
DEBUG:capa.match-function-id:matching function: 0x401ac3
DEBUG:viv_utils.flirt:found library function: 0x401ac3: __setenvp
DEBUG:capa.match-function-id:matching function: 0x401c15
DEBUG:viv_utils.flirt:found local function name: 0x401c15: _parse_cmdline
DEBUG:capa.match-function-id:matching function: 0x401b7c
DEBUG:viv_utils.flirt:found library function: 0x401b7c: __setargv
DEBUG:capa.match-function-id:matching function: 0x402c50
DEBUG:viv_utils.flirt:found local function name: 0x402cc0: ?
DEBUG:viv_utils.flirt:found local function name: 0x402cec: ?
DEBUG:viv_utils.flirt:found local function name: 0x402d10: ?
DEBUG:viv_utils.flirt:found local function name: 0x402d4c: ?
DEBUG:viv_utils.flirt:found local function name: 0x402e48: ?
DEBUG:viv_utils.flirt:found local function name: 0x402e68: ?
DEBUG:viv_utils.flirt:found local function name: 0x402e90: ?
DEBUG:viv_utils.flirt:found local function name: 0x402ee4: ?
DEBUG:viv_utils.flirt:found library function: 0x402c50: ?
DEBUG:capa.match-function-id:matching function: 0x401dc9
DEBUG:viv_utils.flirt:found library function: 0x401dc9: ___crtGetEnvironmentStringsA
DEBUG:capa.match-function-id:matching function: 0x401efb
DEBUG:viv_utils.flirt:found library function: 0x401efb: __ioinit
DEBUG:capa.match-function-id:matching function: 0x40191b
DEBUG:viv_utils.flirt:found local function name: 0x40191b: _fast_error_exit
DEBUG:capa.match-function-id:matching function: 0x40332f
DEBUG:viv_utils.flirt:found library function: 0x40332f: ___sbh_heap_init
DEBUG:capa.match-function-id:matching function: 0x403280
DEBUG:viv_utils.flirt:found library function: 0x403280: _strstr
DEBUG:capa.match-function-id:matching function: 0x4020a6
DEBUG:viv_utils.flirt:found library function: 0x4020a6: __GetLinkerVersion
DEBUG:capa.match-function-id:matching function: 0x4047ad
DEBUG:viv_utils.flirt:found library function: 0x4047ad: __isctype
DEBUG:capa.match-function-id:matching function: 0x4046e1
DEBUG:viv_utils.flirt:found library function: 0x4046e1: ?
DEBUG:capa.match-function-id:matching function: 0x402f9c
DEBUG:viv_utils.flirt:found local function name: 0x402f9c: _strtoxl
DEBUG:capa.match-function-id:matching function: 0x402f85
DEBUG:viv_utils.flirt:found library function: 0x402f85: _strtol
DEBUG:capa.match-function-id:matching function: 0x4031c0
DEBUG:viv_utils.flirt:found library function: 0x4031c0: _strchr
DEBUG:capa.match-function-id:matching function: 0x4020d3
DEBUG:viv_utils.flirt:found library function: 0x4020d3: ___heap_select
DEBUG:capa.match-function-id:matching function: 0x40221b
DEBUG:viv_utils.flirt:found library function: 0x40221b: __heap_init
DEBUG:capa.match-function-id:matching function: 0x401817
DEBUG:viv_utils.flirt:found library function: 0x401817: _mainCRTStartup
DEBUG:capa.match-function-id:matching function: 0x40234e
DEBUG:capa.match-function-id:matching function: 0x4022ba
DEBUG:capa.match-function-id:matching function: 0x404bc8
DEBUG:capa.match-function-id:matching function: 0x402278
DEBUG:viv_utils.flirt:found local function name: 0x402298: __unwind_handler
DEBUG:viv_utils.flirt:found library function: 0x402278: __global_unwind2
DEBUG:viv_utils.flirt:found library function: 0x4022ba: __local_unwind2
DEBUG:viv_utils.flirt:found library function: 0x40234e: __NLG_Notify
DEBUG:capa.match-function-id:matching function: 0x402370
DEBUG:viv_utils.flirt:found library function: 0x402370: __except_handler3
DEBUG:capa.match-function-id:matching function: 0x402298
```